lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -78,6 +78,7 @@ def fetch_bindport
   end
 
   def generate(opts = {})
+    datastore['FETCH_SRVHOST'] = datastore['LHOST'] if datastore['FETCH_SRVHOST'].blank?
```

Comment:
I realized that if you use a bind payload LHOST won't be registered and then by extension `FETCH_SRVHOST` won't be registered. I think a check needs to be added to make sure that they can't both be blank.

Reply:
Looks good to me now, thanks!

```
msf6 exploit(windows/smb/psexec) > set --clear LHOST
LHOST => 
msf6 exploit(windows/smb/psexec) > run

[-] 192.168.159.10:445 - Exploit failed: bad-config: FETCH_SRVHOST required
[*] Exploit completed, but no session was created.
msf6 exploit(windows/smb/psexec) >
```